src/ol/renderer/webgl/shaders.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * @module ol/renderer/webgl/shaders
+ */
+import {asArray} from '../../color.js';
+
+/**
+ * Attribute names used in the default shaders.
+ * @enum {string}
+ */
+export const DefaultAttributes = {
+  COLOR: 'color',
+  OPACITY: 'opacity',
+  WIDTH: 'width',
+};
+
+/**
+ * Packs red/green/blue channels of a color into a single float value; alpha is ignored.
+ * This is how DefaultAttributes.COLOR is expected to be computed.
+ * @param {import("../../color.js").Color|string} color Color as array of numbers or string
+ * @return {number} Float value containing the color
+ * @api
+ */
+export function packColor(color) {
+  const array = asArray(color);
+  const r = array[0] * 256 * 256;
+  const g = array[1] * 256;
+  const b = array[2];
+  return r + g + b;
+}
+
+const DECODE_COLOR_EXPRESSION = `vec3(
+  fract(floor(a_color / 256.0 / 256.0) / 256.0),
+  fract(floor(a_color / 256.0) / 256.0),
+  fract(a_color / 256.0)
+);`;
+
+/**
+ * Default polygon vertex shader.
+ * Relies on DefaultAttributes.COLOR and DefaultAttributes.OPACITY.
+ * @type {string}
+ * @api
```

Comment:
Since this is experimental, I removed this `@api` annotation in an additional commit.